enterprise/frontend/src/embedding-sdk/components/private/InteractiveAdHocQuestion.tsx

```diff
@@ -18,55 +16,29 @@ interface InteractiveAdHocQuestionProps {
 }
 
 export const InteractiveAdHocQuestion = ({
-  isOpenedFromDashboard,
   questionPath,
   onNavigateBack,
   withTitle = true,
   height,
   plugins,
 }: InteractiveAdHocQuestionProps) => {
-  const dispatch = useDispatch();
-
-  const queryResults = useSelector(getQueryResults);
-
-  const [isQuestionLoading, setIsQuestionLoading] = useState(true);
-
-  const loadQuestion = async (
-    dispatch: ReturnType<typeof useDispatch>,
-    questionUrl: string,
-  ) => {
-    setIsQuestionLoading(true);
-
-    const { location, params } = getQuestionParameters(questionUrl);
-    try {
-      await dispatch(initializeQBRaw(location, params));
-    } catch (e) {
-      console.error(`Failed to get question`, e);
-      setIsQuestionLoading(false);
-    }
-  };
-
-  useEffect(() => {
-    loadQuestion(dispatch, questionPath);
-  }, [dispatch, questionPath]);
-
-  useEffect(() => {
-    if (queryResults) {
-      setIsQuestionLoading(false);
-    }
-  }, [queryResults]);
+  const { location, params } = useMemo(
```

Comment:
I'm not sure why these are the sole variables that are left at this level, maybe it might even be more appropriate to pass `questionPath` to the provider and construct these 2 variables inside the `loadQuestion` function as before so we don't have to memoize this.

Reply:
We have a different `getQuestionParameters` in `InteractiveQuestion` that returns `location` and `params` - because the adhoc question can use the QB hash and/or slugs I didn't want to touch it since I'm not 100% sure that the logic will transfer

Reply:
We could pass the function but I feel like that would be super weird